projects/asn1-proto/asn1_proto_converter.cc

```diff
@@ -0,0 +1,169 @@
+#include "asn1_proto_converter.h"
+
+namespace asn1_proto {
+
+// Returns the number of bytes needed to encode |num| into a variable-length
+// unsigned integer with no leading zeros.
+uint8_t ASN1ProtoConverter::GetNumBytes(const size_t num) {
+  for (uint8_t shift = sizeof(num); shift != 0; --shift) {
+    if (((num >> ((shift - 1) * 8)) & 0xFF) != 0) {
+      return shift;
+    }
+  }
+  return 1;
+}
+
+// Converts |num| to a variable-length, big-endian representation and inserts
+// the result into into |encoder_| at |pos|.
+void ASN1ProtoConverter::AppendBytes(const size_t num, const size_t pos) {
+  std::vector<uint8_t> len_vec;
+  for (uint8_t shift = GetNumBytes(num); shift != 0; shift--) {
+    len_vec.push_back((num >> ((shift - 1) * 8)) & 0xFF);
+  }
+  encoder_.insert(encoder_.begin() + pos, len_vec.begin(), len_vec.end());
+}
+
+// If Override Length is set, then this function will set the length to the
+// arbitrary bytes assigned by the proto.
```

Comment:
```suggestion
// Writes |raw_length| to |encoder_| at |len_pos| and returns the number of bytes written.
```

The goal here is to describe with the indicative mood, not the imperative mood (see https://google.github.io/styleguide/cppguide.html#Function_Comments )

The other part is trying to document this function as a stand-alone, based purely on the arguments coming in and the arguments going out. The problem with mentioning "override length" is two-fold:
* there's nothing called override length in the inputs/outputs here
* this is describing how the function /is/ used, not how what it does